lib/content_dumper.rb

```diff
@@ -0,0 +1,51 @@
+require "csv"
+require "zlib"
+
+class ContentDumper
+  FIELDS = %w(base_path content_id locale document_type schema_name rendering_app publishing_app updated_at).freeze
+  HASH_FIELDS = %w(details expanded_links).freeze
+
+  def initialize(filename)
+    @filename = filename
+  end
+
+  def dump
+    Zlib::GzipWriter.open(filename) do |file|
+      csv = CSV.new(file)
+      csv << csv_field_names
+      ContentItem.each do |row|
+        csv << csv_fields(row)
+      end
+    end
+  end
+
+private
+
+  attr_reader :filename
+
+  def csv_field_names
+    FIELDS + HASH_FIELDS.map { |field| "#{field}_hash".to_sym }
+  end
+
+  def csv_fields(item)
+    FIELDS.map { |field| item.send(field) } + HASH_FIELDS.map { |field| hash_field(item.send(field)) }
+  end
+
+  def hash_field(object)
+    Digest::MD5.hexdigest(
+      JSON.generate(
+        deep_sort_hash(object)
```

Comment:
looking at Pub API these are sorted: https://github.com/alphagov/publishing-api/blob/master/app/queries/edition_links_from.rb#L29 so I think we can probably simplify this down to just running the JSON.generate on it.

Might worth adding a note about how the more expected `.to_json` wasn't used as we get more consistent results with JSON.generate

Also did you get a speed boost with MD5? I was just thinking that for someone reading the use of that over SHA1 for instance might cause alarm bells so if we're getting extra performance that should be noted in commit or even a comment and if it's negligible might want to use a SHA one?